config/bot_filter.go

```diff
@@ -0,0 +1,47 @@
+package config
+
+import (
+	"golang.ngrok.com/ngrok/internal/pb"
+)
+
+// BotFilter is a pair of strings slices that allow/deny traffic to an endpoint
+// and 2 boolean flags that either allow empty user agents or use a
+// ngrok defined deny list
+type botFilter struct {
+	// slice of regex strings for allowed user agents
+	Allow []string
+	// slice of regex strings for denied user agents
+	Deny []string
+	// description of what users have defined
+	Description string
+	// allows for empty user agents traffic to make it endpoint
+	AllowEmptyUserAgent bool
+	// uses internal list of commonly known bots
+	UseDefaultDeny bool
+}
+
+func (b *botFilter) toProtoConfig() *pb.MiddlewareConfiguration_BotFilter {
+	if b == nil {
+		return nil
+	}
+	return &pb.MiddlewareConfiguration_BotFilter{
+		Allow:               b.Allow,
+		Deny:                b.Deny,
+		Description:         b.Description,
+		AllowEmptyUserAgent: b.AllowEmptyUserAgent,
+		UseDefaultDeny:      b.UseDefaultDeny,
+	}
+}
+
+// WithBotFilter configures botfilter from a set of passed parameters.
```

Comment:
This doc comment is missing an explanation of what each of the arguments do since it's not obvious (or link to other docs if we can)

The `botFilter` struct isn't exported, so the comments over its various fields aren't visible in the user-facing docs, so these arguments are the only place we get to explain them.